lib/extract.js

```diff
@@ -17,8 +17,17 @@ function Extract (opts) {
 
   self.on('entry', function(entry) {
     if (entry.type == 'Directory') return;
+
+    // to avoid zip slip (writing outside of the destination), we resolve
+    // the target path, and make sure it's nested in the intended
+    // destination, or not extract it otherwise.
+    var extractPath = path.join(opts.path, entry.path);
+    if (extractPath.indexOf(opts.path) != 0) {
+      return;
```

Comment:
This should either throw an error or at the very least show a warning.
Also the `entry` is left hanging here as nothing drains it  (possibly use .autodrain)